components/tournament_ticker/tournament_ticker.lua

```diff
@@ -0,0 +1,200 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Tournament/Ticker
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local DateExt = require('Module:Date/Ext')
+local Lpdb = require('Module:Lpdb')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Table = require('Module:Table')
+local Tier = require('Module:Tier/Utils')
+
+local TournamentTicker = {}
+
+---@class StandardTournament
+---@field displayName string
+---@field pageName string
+---@field startDate {year: integer, month: integer?, day: integer?, timestamp: integer?}?
+---@field endDate {year: integer, month: integer?, day: integer?, timestamp: integer?}?
+---@field liquipediaTier string
+---@field liquipediaTierType string
+---@field region string
+---@field featured boolean
+---@field status 'UPCOMING'|'ONGOING'|'FINISHED'
+
+---@class TournamentStatus
+---@field isTournamentInStatus fun(record: StandardTournament): boolean
+---@field enum 'UPCOMING'|'ONGOING'|'FINISHED'
+
+local TOURNAMENT_STATUS = {
+	UPCOMING = 'UPCOMING',
+	ONGOING = 'ONGOING',
+	FINISHED = 'FINISHED',
+}
+
+---@type TournamentStatus
+local STATUS_UPCOMING = {
+	enum = TOURNAMENT_STATUS.UPCOMING,
+	isTournamentInStatus = function(tournament)
+		--- TODO return false if before threshold
+		return tournament.startDate.timestamp > DateExt.getCurrentTimestamp()
+	end,
+	sort = function(tournament1, tournament2)
+		if tournament1.startdate ~= tournament2.startdate then
+			return tournament1.startdate > tournament2.startdate
+		end
+		return tournament1.sortdate > tournament2.sortdate
+	end
+}
+---@type TournamentStatus
+local STATUS_ONGOING = {
+	enum = TOURNAMENT_STATUS.ONGOING,
+	isTournamentInStatus = function(tournament)
+		return tournament.sortdate ~= DateExt.defaultDate and tournament.endDate.timestamp < DateExt.getCurrentTimestamp()
+	end,
+	sort = function(tournament1, tournament2)
+		if tournament1.sortdate ~= tournament2.sortdate then
+			return tournament1.sortdate < tournament2.sortdate
+		end
+		return tournament1.startdate > tournament2.startdate
+	end
+}
+
+---@type TournamentStatus
+local STATUS_CONCLUDED = {
+	enum = TOURNAMENT_STATUS.FINISHED,
+	isTournamentInStatus = function(tournament)
+		--- TODO return false if after threshold
+		return tournament.startDate.timestamp < DateExt.getCurrentTimestamp()
+			and tournament.endDate.timestamp > DateExt.getCurrentTimestamp()
+	end,
+	sort = function(tournament1, tournament2)
+		if tournament1.sortdate ~= tournament2.sortdate then
+			return tournament1.sortdate < tournament2.sortdate
+		end
+		return tournament1.startdate > tournament2.startdate
+	end
+}
+
+function TournamentTicker.getTournamentsFromDB()
+	local tournaments = {}
+	Lpdb.executeMassQuery('tournament', {}, function (record)
```

Comment:
we really should add conditions here!
- exclude cancelled events?!
- use the supplied thresholds to limit the results
- option for custom conditions (as args input?)


else you run through all tournaments which seems inefficient af
and it also includes unwanted ones

Reply:
support added

Reply:
did some memory optimzation. But yeah I was thinking about a +/- 365 before

Reply:
well some wikis have >>20k tournaments, the lpdb query time alone is not nice for that already

Reply:
Easy to readjust, this should be very fast, and if it ends up being problematic we know what to do